web/src/lib/components/validator.tsx

```diff
@@ -21,17 +21,29 @@ interface State {
 /**
  * Widget for validating voice clips.
  */
-export default class Validator extends Component<Props, State> {
+export default class Validator extends React.Component<Props, State> {
+  state = { loading: false, glob: '', sentence: '', audioSrc: '' };
+  private _isMounted = false;
```

Comment:
Any reason why do we need this? It seems a bit weird to me

Reply:
The `_isMounted`-part?

Reply:
yeah

Reply:
`Component.isMounted` was deprecated a few React versions ago, because it is a bit of an anti-pattern to check it. But calling `setState` after a Component was unmounted is even worse, aka illegal. Unfortunately in this case we don't have a way to cancel the promise, so we had to go with the check. Preact was just very lenient about where `setState` can be called,.
I'm open for recommendations on how we could get around this though :)

Reply:
Ah I see, well, I think this current code style is what we can live with.